lib/active_model_serializers/lookup_chain.rb

```diff
@@ -0,0 +1,80 @@
+module ActiveModelSerializers
+  module LookupChain
+    # Standard appending of Serializer to the resource name.
+    #
+    # Example:
+    #   Author => AuthorSerializer
+    BY_RESOURCE = lambda do |resource_class, _serializer_class, _namespace|
+      serializer_from(resource_class)
+    end
+
+    # Uses the namespace of the resource to find the serializer
+    #
+    # Example:
+    #  British::Author => British::AuthorSerializer
+    BY_NAMESPACE = lambda do |resource_class, _serializer_class, _namespace|
```

Comment:
I think it would be more comprehensible and accurate to call this `BY_RESOURCE_NAMESPACE` and the one below `BY_NAMESPACE`. `BY_RESOURCE_NAMESPACE` makes it clear that we use the namespace from the resource. Otherwise it may be ambiguous. Also the lookup below may not be the "controller" namespace but any kind of namespace. For example, we used to have all the serializer in a `Serializer` module e.g. `Serializer::V1::AuthorSerializer`, `Serializer::V2::AuthorSerializer`.